File: LHDS.AdminPortal.Web/ClientApp/src/models/dataSetSpecifications/dataSetSpecification.ts
import { Guid } from 'guid-typescript';
import { DataSet } from '../dataSets/dataSet';

export class DataSetSpecification {
    public id: Guid;
    public dataSetId: Guid;
    public supplierSpecificationVersion: string;
    public ourSpecificationVersion: string;
    public notes: string;
    public isMultiSender: boolean;
    public entityChangeSynchronisation: string;
    public dateReleased?: Date;
    public dateImplemented?: Date;
    public dateSuperseded?: Date;
    public presededBy: string;
    public supersededBy: string;
    public isPublished: boolean;
    public isActive: boolean;
    public activeFrom: Date;
    public activeTo: Date;
    public createdBy?: string;
    public createdDate?: Date;
    public updatedBy?: string;
    public updatedDate?: Date;
    public dataSet?: DataSet;

    constructor(dataSetSpecification: any) {
        this.id = dataSetSpecification.id ? Guid.parse(dataSetSpecification.id) : Guid.parse(Guid.EMPTY);
        this.dataSetId = dataSetSpecification.dataSetId;
        this.supplierSpecificationVersion = dataSetSpecification.name;
        this.ourSpecificationVersion = dataSetSpecification.ourSpecificationVersion;
        this.notes = dataSetSpecification.notes;
        this.isMultiSender = dataSetSpecification.isMultiSender;
        this.entityChangeSynchronisation = dataSetSpecification.entityChangeSynchronisation;
        this.dateReleased = dataSetSpecification.dateReleased;
        this.dateImplemented = dataSetSpecification.dateImplemented;
        this.dateSuperseded = dataSetSpecification.dateSuperseded;
        this.presededBy = dataSetSpecification.presededBy;
        this.supersededBy = dataSetSpecification.dataSourceType;
        this.isPublished = dataSetSpecification.isPublished;
        this.isActive = dataSetSpecification.isActive;
        this.activeFrom = dataSetSpecification.activeFrom;
        this.activeTo = dataSetSpecification.activeTo;
        this.createdBy = dataSetSpecification.createdBy;
        this.createdDate = new Date(dataSetSpecification.createdDate);
        this.updatedBy = dataSetSpecification.updatedBy;
        this.updatedDate = new Date(dataSetSpecification.updatedDate);

        if (dataSetSpecification.dataSet !== undefined && dataSetSpecification.dataSet !== null) {
            this.dataSet = new DataSet(dataSetSpecification.dataSet);
        }
    }
}

File: LHDS.ConfigImportExportTool/Models/Foundations/DatasetSpecifications/DatasetSpecification.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;
using System.Collections.Generic;
using LHDS.ConfigImportExportTool.Models.Bases;
using LHDS.ConfigImportExportTool.Models.Foundations.Datasets;
using LHDS.ConfigImportExportTool.Models.Foundations.SpecificationObjects;

namespace LHDS.ConfigImportExportTool.Models.Foundations.DatasetSpecifications
{
    public class DataSetSpecification : IKey, IAudit
    {
        public Guid Id { get; set; }
        public Guid DataSetId { get; set; }
        public string SupplierSpecificationVersion { get; set; } = string.Empty;
        public string OurSpecificationVersion { get; set; } = string.Empty;
        public string Notes { get; set; } = string.Empty;
        public bool IsMultiAuthorPerBatch { get; set; }
        public string EntityChangeSynchronisation { get; set; } = string.Empty;
        public DateTimeOffset? DateReleased { get; set; }
        public DateTimeOffset? DateImplemented { get; set; }
        public DateTimeOffset? DateSuperseded { get; set; }
        public Guid? SupersededById { get; set; }
        public Guid? PresededById { get; set; }
        public bool IsPublished { get; set; }
        public bool IsActive { get; set; }
        public DateTimeOffset ActiveFrom { get; set; }
        public DateTimeOffset ActiveTo { get; set; }
        public string CreatedBy { get; set; } = string.Empty;
        public string UpdatedBy { get; set; } = string.Empty;
        public DateTimeOffset UpdatedDate { get; set; }
        public DateTimeOffset CreatedDate { get; set; }
        public DataSet DataSet { get; set; } = null!;
        public List<SpecificationObject> SpecificationObjects { get; set; } = new List<SpecificationObject>();
        public List<DataSetSpecification> SupersededBy { get; set; } = new List<DataSetSpecification>();
        public List<DataSetSpecification> PresededBy { get; set; } = new List<DataSetSpecification>();
    }
}
